x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/controls/actions.tsx

```diff
@@ -71,58 +93,96 @@ export const Actions = ({
   onInvestigateInTimeline,
   onSearchToggle,
   searchFilterCounter = 0,
+  searchToggled,
   ...props
 }: ActionsProps) => {
   const { euiTheme } = useEuiTheme();
-  const [searchToggled, setSearchToggled] = useState<boolean>(false);
+  const [isSearchBarTourOpen, setIsSearchBarTourOpen] = useState(false);
+  const [shouldShowSearchBarButtonTour, setShouldShowSearchBarButtonTour] = useLocalStorage(
+    SHOW_SEARCH_BAR_BUTTON_TOUR_STORAGE_KEY,
+    true
+  );
+
+  useEffect(() => {
```

Comment:
as we only rely on props and state here, can we avoid using useEffect here?
https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state